testutil/integration/suites/update_params.go

```diff
@@ -0,0 +1,424 @@
+//go:build integration
+
+package suites
+
+import (
+	"encoding/hex"
+	"reflect"
+	"testing"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/app/volatile"
+	"github.com/pokt-network/poktroll/testutil/cases"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+	tokenomicstypes "github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+// ParamType is a type alias for a module parameter type. It is the string that
+// is returned when calling reflect.Value#Type()#Name() on a module parameter.
+type ParamType = string
+
+const (
+	ParamTypeInt64   ParamType = "int64"
+	ParamTypeUint64  ParamType = "uint64"
+	ParamTypeFloat32 ParamType = "float32"
+	ParamTypeString  ParamType = "string"
+	ParamTypeBytes   ParamType = "uint8"
+	ParamTypeCoin    ParamType = "Coin"
+)
+
+const (
+	MsgUpdateParamsName = "MsgUpdateParams"
+	MsgUpdateParamName  = "MsgUpdateParam"
+)
+
+// ModuleParamConfig holds a set of valid parameters and type information for a
+// given module. It uses any instead of cosmostypes.Msg to mitigate easy mistakes
+// that could result from using pointers instead (unintended mutation). It is still
+// possible to mutate this global variable; however, it is less likely to happen
+// unintentionally.
+type ModuleParamConfig struct {
+	ValidParams             any
+	MsgUpdateParams         any
+	MsgUpdateParamsResponse any
+	MsgUpdateParam          any
+	MsgUpdateParamResponse  any
+	ParamTypes              map[ParamType]any
+	QueryParamsRequest      any
+	QueryParamsResponse     any
+	DefaultParams           any
+	NewParamClientFn        any
+}
+
+var (
+	// AuthorityAddr is the cosmos account address of the authority for the integration app.
+	AuthorityAddr cosmostypes.AccAddress
+	// AuthorizedAddr is the cosmos account address which is the grantee of authz
+	// grants for parameter update messages.
+	AuthorizedAddr cosmostypes.AccAddress
+
+	ValidServiceFeeCoin                = cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 1000000001)
+	ValidMissingPenaltyCoin            = cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 500)
+	ValidSubmissionFeeCoin             = cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 5000000)
+	ValidRelayDifficultyTargetHash, _  = hex.DecodeString("00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
+	ValidProofRequirementThresholdCoin = cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 100)
+
+	// ModuleParamConfigMap is a map of module names to their respective parameter
```

Comment:
Does this involve updating the documentation in [1] as well?

https://dev.poktroll.com/develop/developer_guide/adding_params?_highlight=adding

Reply:
TODO_DOCUMENT?